core/drivers/tpm2/tpm2_chip.c

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, Linaro Limited
+ */
+
+#include <assert.h>
+#include <io.h>
+#include <trace.h>
+#include <drivers/tpm2_chip.h>
+#include <tpm2.h>
+
+struct tpm2_chip *tpm2_device;
+
+enum tpm2_result tpm2_chip_send(uint8_t *buf, uint32_t len)
+{
+	if (!tpm2_device || !tpm2_device->ops)
```

Comment:
`tpm2_chip_register()` already checks `tpm2_device->ops != NULL`.
Maybe it should verify `.send` and `.recv` are populated?